CMakeLists.txt

```diff
@@ -159,10 +159,16 @@ elseif(HTTPLIB_USE_BROTLI_IF_AVAILABLE)
 endif()
 
 if(HTTPLIB_REQUIRE_ZSTD)
-	find_package(zstd REQUIRED)
+	find_package(zstd)
+	if(NOT zstd_FOUND)
+		find_package(PkgConfig REQUIRED)
+		pkg_check_modules(zstd REQUIRED IMPORTED_TARGET libzstd)
+		add_library(zstd::libzstd ALIAS PkgConfig::zstd)
+	endif()
 	set(HTTPLIB_IS_USING_ZSTD TRUE)
 elseif(HTTPLIB_USE_ZSTD_IF_AVAILABLE)
 	find_package(zstd QUIET)
+	# TODO: try to find it via pkg-config
 	set(HTTPLIB_IS_USING_ZSTD ${zstd_FOUND})
```

Comment:
I suppose something like this could work for your TODO.
```cmake
find_package(zstd QUIET)
	if(NOT zstd_FOUND)
		find_package(PkgConfig QUIET)
		if(PKG_CONFIG_FOUND)
			pkg_check_modules(zstd QUIET IMPORTED_TARGET libzstd)

			if(TARGET PkgConfig::zstd)
				add_library(zstd::libzstd ALIAS PkgConfig::zstd)
			endif()
		endif()
		# Both find_package and PkgConf seem to set a XXX_FOUND var
		set(HTTPLIB_IS_USING_ZSTD ${zstd_FOUND})
	endif()
```

It's a shame there's not some find_package+PkgConf combo thing for Cmake. This seems to be a common use-case..

Reply:
Yeah, something like that. For now I'll just copy your code here. I didn't write that because I don't need it for my use case, and it's quite long. If user didn't specify they want zstd, they would probably be satisfied if it's not found even if it could be found.

In another project I handled the same problem differently: https://github.com/znc/znc/blob/df6a5073ff9a68bc34981c999ec9ad1e0c4bac09/CMakeLists.txt#L71-L79 and usage of it is visible below, where `${TRISTATE_ARGON_REQUIRED}` and such are passed to `find_package`/`pkg_check_modules` in place of `REQUIRED`. The trick is that `AUTO` evaluates to true in `if(WANT_FOO)`.

I you follow a similar pattern here, at least can dedup code which requires the dep and which doesn't require it.